src/main/java/io/divolte/server/IncomingRequestProcessor.java

```diff
@@ -150,14 +154,19 @@ public ProcessingDirective process(final DivolteEvent event) {
 
             if (!duplicate || keepDuplicates) {
                 final GenericRecord avroRecord = mapper.newRecordFromExchange(event);
-                final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
+                try {
+                    final AvroRecordBuffer avroBuffer = AvroRecordBuffer.fromRecord(
                         event.partyCookie,
                         event.sessionCookie,
                         event.requestStartTime,
                         event.clientUtcOffset,
                         avroRecord);
-                listener.incomingRequest(event, avroBuffer, avroRecord);
-                doProcess(avroBuffer);
+                    listener.incomingRequest(event, avroBuffer, avroRecord);
+                    doProcess(avroBuffer);
+                } catch (final RuntimeException e) {
```

Comment:
The listener call and the doProcess should go outside of the try-catch block, if we're guarding against mapping errors here.